### PR TITLE
Modal hooks

### PR DESCRIPTION
**Pull request by JoshRosenstein**
# Summary
Updated [Internals/Modal](https://github.com/contiamo/operational-ui/blob/master/src/Internals/Modal.tsx)  to use hooks.
Also,made small change on its type and the state model since options was the only thing stored.

# Related issue
[910](https://github.com/contiamo/operational-ui/issues/910)

# To be tested

Me
- [x] No error or warning in the console on `localhost:6060`

Tester 1

- [ ] Things look good on the demo.
  

Tester 2

- [ ] Things look good on the demo.

### CODE REVIEW

File: src/Internals/Modal.tsx [hunk 1]
```diff
@@ -8,47 +7,33 @@ export interface ModalOptions {
...
 }
 
-export interface State {
-  options: Partial<ModalOptions>
-}
+export type State = Partial<ModalOptions>
 
 export interface Props {
-  children: (modal: (options: ModalOptions) => void) => React.ReactNode
+  children: (modal: (options: ModalOptions) => void) => JSX.Element
```
Comment (TejasQ) [inline src/Internals/Modal.tsx]: I'm not sure this is correct? We want `children` to return a ReactNode...

**Comment by JoshRosenstein** [comment]
@TejasQ - i was also thinking of that- but couple things
 -wanted to "upgrade" this component before potentially making a breaking changes
But there are also several different options to tackle this. 
Heres some reads samples
[kentcdodds](https://kentcdodds.com/blog/react-hooks-whats-going-to-happen-to-render-props)
[the hydra pattern](https://americanexpress.io/hydra/)

File: src/Internals/Modal.tsx (hunk 1 above)
Comment (JoshRosenstein) [reply src/Internals/Modal.tsx]: same issue as Tabs - will have to keep as JSX.Element | null , unless there's another solution im missed

Comment (JoshRosenstein) [reply src/Internals/Modal.tsx]: So i opened this back up and ran start- It didnt throw type error when using ReactNode.
So I think for the other ones that throw a type error  - When I wrap the output in a fragment like Tabs is, it wont throw. Do you prefer changing the type, or wrapping in a <></>

**Comment by JoshRosenstein** [comment]
Some current methods in the wild
[react-modal-hook](https://github.com/mpontus/react-modal-hook)-[Example](https://codesandbox.io/s/v8qy4w1j77)
[ProtonMail](https://github.com/ProtonMail/react-components/tree/master/components/modal)
[react-use-modal](https://github.com/wowlusitong/react-use-modal/tree/master)- [Example](https://react-use-modal.netlify.com/)
[neinteractiveliterature/intercode](https://github.com/neinteractiveliterature/intercode/tree/master/app/javascript/ModalDialogs)

File: src/Internals/Modal.tsx [hunk 2]
```diff
@@ -8,47 +7,33 @@ export interface ModalOptions {
...
-              : this.state.options.body}
-          </ControlledModal>
-        )}
-      </>
-    )
-  }
+export const Modal = ({ children }: Props) => {
+  const [options, setOptions] = useState<State>({})
+
+  const openModal = useCallback((newoptions: ModalOptions) => {
```
Comment (stereobooster) [inline src/Internals/Modal.tsx]: `newOptions`

**Comment by stereobooster** [comment]
@JoshRosenstein one small fix of variable name otherwise clear and precise 👍

File: src/Internals/Modal.tsx (hunk 2 above)
Comment (JoshRosenstein) [reply src/Internals/Modal.tsx]: 🤦‍♂️ 😆

**Comment by JoshRosenstein** [comment]
Should open an issue after this merges to continue discussion of potential options for better implementation.

**Comment by stereobooster** [comment]
@TejasQ answering `useModal` question, but on the other side "modal" is a component (I know it is possible to return component from hook or create component outside of the current react tree) I prefer components for anything that have visual representation.

My question - this is not really a feature. I meant this PR doesn't add feature. Do we have (or want to add) `**Chore**` tag? For refactoring and other non project related things?

**Pull request merged**